salesforcecli/plugin-source
# Sm/source-tracking-everywhere

### What does this PR do?
add `--tracksource` and `--forceoverwrite` for deploy,retrieve,delete
centralize conflict checks
centralize tracking file update logic to be shared
NUT for happy path of these commands

### What issues does this PR fix or reference?
@W-9914352@

### QA suggestions

- Look at the NUT and explore other scenario for deploy/retrieve/delete
- Test with some of our least favorite mdTypes (labels, COFT, etc)
- try a destructive deploy (with manifests) because I didn't...verify that updates tracking files correctly
- do some of the basics for push/pull, too (because their tracking file interactions changed to make them shared across the functions)
- check all the help formatting and examples
